README.md
<h1 align='center'><img src='https://cdn3.iconfinder.com/data/icons/bold-blue-glyphs-free-samples/32/Info_Circle_Symbol_Information_Letter-64.png' width='30' /> Info System</h1>

<p align='center'>Info-system is a library that provides you see basic information about your Operational System in a beautiful way!</p>

# Install

```node
<<<<<<< HEAD
$ npm i -g info-system
=======
$ npm i info-system -g
>>>>>>> c5ede0934f7488e1ba515c9a82340bd9a72e204f
```

# Usage

<p>Currently have just a few commands line:</p>

To show all infos

```node
$ info-system
```

<hr>

Show only memory infos

```node
$ info-system --memory
```

or

```node
$ info-system -m
```

<hr>

Show only processor infos

```node
$ info-system --processor
```

or

```node
$ info-system -p
```

<hr>

Show only OS infos

```node
$ info-system --operational
```

or

```node
$ info-system -os
```

<br />

### You can see here how it stay

<br />

<img src='./assets/info-system.gif' />

<br />
<br />

## 🐛 Bug Report

If you find an issue, report here <https://github.com/Juniordell/info-system/issues>, you'll help a lot!


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)

## [2.1.1] - 2017-06-20

### Fixed

- Fix README.md bug

## [2.1.1] - 2017-06-20

### Added

- CHANGELOG.md
- Info about the current time

### Changed

- The titles of each table became better

## [2.1.0] - 2020-09-07

### Changed

- Interface become better for the user

[2.1.2]: https://www.npmjs.com/package/info-system/v/2.1.1
[2.1.1]: https://www.npmjs.com/package/info-system/v/2.1.1
[2.1.0]: https://www.npmjs.com/package/info-system/v/2.1.0


calc/memory.js
const os = require("os");
const bytes = require("bytes");
const sys = require("systeminformation");

const totalRAMMem = bytes(os.totalmem());
const freeRAMMem = bytes(os.freemem());

async function getTotalDiskMemory() {
  const disks = await sys.fsSize();
  let total = 0;

  for (disk of disks) {
    if (disk.size) {
      total += disk.size;
    }
  }

  return bytes(total);
}

let totalDiskMem = getTotalDiskMemory()
  .then((result) => (totalDiskMem = result))
  .catch((err) => console.log(err));

async function getFreeDiskMemory() {
  const disks = await sys.fsSize();
  let totalUsed = 0;
  let total = 0;

  for (disk of disks) {
    if (disk.used) {
      totalUsed += disk.used;
    }
    if (disk.size) {
      total += disk.size;
    }
  }

  return bytes(total - totalUsed);
}

let freeDiskMem = getFreeDiskMemory()
  .then((result) => (freeDiskMem = result))
  .catch((err) => console.log(err));

module.exports = {
  totalRAMMem,
  freeRAMMem,
  totalDiskMem,
  freeDiskMem,
};


info-system.js
#!/usr/bin/env node

const os = require("os");
const chalk = require("chalk");
const Table = require("cli-table3");
const hourToString = require("./calc/uptime");
const { totalMem, freeMem } = require("./calc/memory");
const formattedSpeed = require("./calc/speed");

const table = new Table({
  head: ["Info", "Results"],
  colWidths: [25, 37],
});

let platform;

switch (os.platform()) {
  case "darwin":
    platform = "Darwin OS ";
    break;
  case "freebsd":
    platform = "FreeBSD ";
    break;
  case "linux":
    platform = "Linux ";
    break;
  case "sunos":
    platform = "SunOS ";
    break;
  case "win32":
    platform = "Windows ";
    break;
}

table.push(
  ["Total Memory", chalk.blue(totalMem)],
  ["Free Memory", chalk.greenBright(freeMem)],
  ["Host Name", chalk.yellow(os.hostname())],
  ["Platform", chalk.blueBright(platform)],
  ["OS Architecture", chalk.magentaBright(os.arch())],
  ["OS Type", chalk.cyanBright(os.type())],
  ["System Uptime", chalk.redBright(hourToString)],
  ["Processor", chalk.rgb(290, 133, 57)(os.cpus()[0].model.trim())],
  ["Processor Speed", chalk.rgb(53, 198, 21)(formattedSpeed)]
);

function showInfos() {
  console.log(table.toString());
}

module.exports = showInfos;


index.js
const os = require("os");
const chalk = require("chalk");
const Table = require("cli-table3");
const cliSpinners = require("cli-spinners");
const ora = require("ora");
const hourToString = require("./calc/uptime");
const {
  totalDiskMem,
  freeDiskMem,
  totalRAMMem,
  freeRAMMem,
} = require("./calc/memory");
const formattedSpeed = require("./calc/speed");
const moment = require("moment");

const now = moment().format("MMMM Do YYYY, h:mm:ss a");

const tableMem = new Table({
  head: ["Infos", "Results"],
  colWidths: [25, 37],
});

const tableProcessor = new Table({
  head: ["Infos", "Results"],
  colWidths: [25, 37],
});

const tableOS = new Table({
  head: ["Infos", "Results"],
  colWidths: [25, 37],
});

function log(any = "") {
  console.log(any);
}

let platform;

switch (os.platform()) {
  case "darwin":
    platform = "Darwin OS ";
    break;
  case "freebsd":
    platform = "FreeBSD ";
    break;
  case "linux":
    platform = "Linux ";
    break;
  case "sunos":
    platform = "SunOS ";
    break;
  case "win32":
    platform = "Windows ";
    break;
}

async function showInfos() {
  const spinner = new ora({
    text: "Fetching your System Data",
    spinner: cliSpinners.random,
  });

  spinner.start();

  setTimeout(() => {
    spinner.color = "blue";
    spinner.text = "Successfully Fetched the System Data";
  }, 2000);

  try {
    await new Promise((resolve) => setTimeout(resolve, 3000));
    spinner.stop();
    totalDiskMem
      .then((result) => {
        tableMem.push(["Total Disk Memory", chalk.blueBright(result)]);

        freeDiskMem
          .then((result) => {
            tableMem.push(
              ["Free Disk Memory", chalk.greenBright(result)],
              ["Total RAM Memory", chalk.blueBright(totalRAMMem)],
              ["Free RAM Memory", chalk.greenBright(freeRAMMem)]
            );

            log();
            log("=".repeat(65));
            log();
            log(" ".repeat(15) + chalk.greenBright(now));
            log();
            log("=".repeat(65));
            log();

            log();
            log(chalk.bold("                    >>> Memory Infos <<<       "));
            log("             ──────────────────────────────────");
            log(tableMem.toString());

            log();

            showProcessor();

            log();
            showOS();
          })
          .catch((err) => log(err));
      })
      .catch((err) => log(err));
  } catch (err) {
    spinner.stop();
    log(err);
  }
}

async function showMem() {
  const spinner = new ora({
    text: "Fetching your System Data",
    spinner: cliSpinners.random,
  });

  spinner.start();

  setTimeout(() => {
    spinner.color = "blue";
    spinner.text = "Successfully Fetched the System Data";
  }, 2000);

  try {
    await new Promise((resolve) => setTimeout(resolve, 3000));
    spinner.stop();
    totalDiskMem
      .then((result) => {
        tableMem.push(["Total Disk Memory", chalk.blueBright(result)]);

        freeDiskMem
          .then((result) => {
            tableMem.push(
              ["Free Disk Memory", chalk.greenBright(result)],
              ["Total RAM Memory", chalk.blueBright(totalRAMMem)],
              ["Free RAM Memory", chalk.greenBright(freeRAMMem)]
            );
            log();
            log(chalk.bold("                    >>> Memory Infos <<<       "));
            log("             ──────────────────────────────────");
            log(tableMem.toString());
          })
          .catch((err) => log(err));
      })
      .catch((err) => log(err));
  } catch (err) {
    spinner.stop();
    log(err);
  }
}

function showProcessor() {
  tableProcessor.push(
    ["Processor", chalk.rgb(290, 133, 57)(os.cpus()[0].model.trim())],
    ["Processor Speed", chalk.rgb(290, 133, 57)(formattedSpeed)],
    ["Cores", chalk.rgb(290, 133, 57)(os.cpus().length)]
  );
  log();
  log(chalk.bold("                  >>> Processor Infos <<<     "));
  log("             ─────────────────────────────────");
  log(tableProcessor.toString());
}

function showOS() {
  tableOS.push(
    ["Host Name", chalk.yellow(os.hostname())],
    ["Platform", chalk.yellowBright(platform)],
    ["OS Architecture", chalk.yellowBright(os.arch())],
    ["OS Type", chalk.yellowBright(os.type())],
    ["System Uptime", chalk.yellowBright(hourToString)]
  );
  log();
  log(chalk.bold("                     >>> OS Infos <<<        "));
  log("             ────────────────────────────────");
  log(tableOS.toString());
}

module.exports = {
  showInfos,
  showMem,
  showProcessor,
  showOS,
  showMem,
};
